plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeMetaModel.java

```diff
@@ -137,4 +137,65 @@ public DBCQueryTransformer createQueryTransformer(@NotNull DBCQueryTransformType
         }
         return null;
     }
+
+    @Override
+    public JDBCStatement prepareUniqueConstraintsLoadStatement(
+        @NotNull JDBCSession session,
+        @NotNull GenericStructContainer owner,
+        @Nullable GenericTableBase forParent
+    ) throws SQLException {
+        boolean recognizeWildCards = isRecognizeWildCards(session, owner);
+        GenericSchema schema = owner.getSchema();
+        String schemaName = getSchemaNameForPattern(session, recognizeWildCards, schema);
+        String tableName = getTableNameForPattern(session, owner, forParent, recognizeWildCards);
+        return session.getMetaData()
+            .getPrimaryKeys(owner.getCatalog() == null ? null : owner.getCatalog().getName(), schemaName, tableName)
+            .getSourceStatement();
+    }
+
+    @Override
+    public JDBCStatement prepareForeignKeysLoadStatement(
+        @NotNull JDBCSession session,
+        @NotNull GenericStructContainer owner,
+        @Nullable GenericTableBase forParent
+    ) throws SQLException {
+        boolean recognizeWildCards = isRecognizeWildCards(session, owner);
+        GenericSchema schema = owner.getSchema();
+        String schemaName = getSchemaNameForPattern(session, recognizeWildCards, schema);
+        String tableName = getTableNameForPattern(session, owner, forParent, recognizeWildCards);
+        return session.getMetaData()
+            .getImportedKeys(owner.getCatalog() == null ? null : owner.getCatalog().getName(), schemaName, tableName)
+            .getSourceStatement();
+    }
+
+    private boolean isRecognizeWildCards(@NotNull JDBCSession session, @NotNull GenericStructContainer owner) throws SQLException {
+        // Snowflake driver do not recognize wild cards patterns before version 3.13.19 - and 19 here is the number of patch, not minor
+        if (owner.getDataSource().isDriverVersionAtLeast(3, 13)) {
+            String driverVersion = session.getMetaData().getDriverVersion();
+            if (CommonUtils.isNotEmpty(driverVersion) && driverVersion.contains(".")) {
+                String[] strings = driverVersion.split("\\.");
+                if (strings.length == 3) {
+                    return CommonUtils.toLong(strings[2]) >= 19;
+                }
+            }
```

Comment:
I know it's temporary, but maybe move this method to the data source and make its signature look like:
```java
boolean isDriverVersionAtLeast(int major, int minor, int patch)
```

?

Reply:
I think it is overcoding. I will move this method if I have another place to use it.